src/main/scala/scorex/core/network/peer/InMemoryPeerDatabase.scala

```diff
@@ -26,17 +26,45 @@ final class InMemoryPeerDatabase(settings: NetworkSettings, timeProvider: TimePr
     */
   private var penaltyBook = Map.empty[InetAddress, (Int, Long)]
 
-  override def get(peer: InetSocketAddress): Option[PeerInfo] = peers.get(peer)
+  override def get(peer: InetSocketAddress): Option[PeerInfo] = peers.get(peer).map(_._1)
+
+  private def addPeer(peerInfo: PeerInfo): Unit = {
+    peerInfo.peerSpec.address.foreach { address =>
+      log.debug(s"Updating peer info for $address")
+      peers += address -> (peerInfo, timeProvider.time())
+    }
+  }
 
   override def addOrUpdateKnownPeer(peerInfo: PeerInfo): Unit = {
     if (!peerInfo.peerSpec.declaredAddress.exists(x => isBlacklisted(x.getAddress))) {
-      peerInfo.peerSpec.address.foreach { address =>
-        log.debug(s"Updating peer info for $address")
-        peers += address -> peerInfo
-      }
+      addPeer(peerInfo)
```

Comment:
`addOrUpdateKnownPeer` is used in 2 cases:
- we are adding peers from the config file
- we have connected to a peer and we are updating peer info in the DB

Not limiting it here is because both these flows are high-priority, and we probably don't want to hit a ceiling here.
So, yes, it can happen that the map will be bigger than the limit, but with the clean-up mechanism(5 sec schedule to try connect to new peer) it should not cause issues

Reply:
will add an in code comment to describe why it's safe here